src/layout/workspace.rs

```diff
@@ -2101,6 +2269,58 @@ impl<W: LayoutElement> Workspace<W> {
             })
     }
 
+    fn insert_hint_area(&self, insert_hint: &InsertHint) -> Option<Rectangle<f64, Logical>> {
+        let mut hint_area = match insert_hint.position {
+            InsertPosition::NewColumn(column_index) => {
+                if column_index == 0 || column_index == self.columns.len() {
```

Comment:
Hint at index 0 is rendered one gap further to the right than it should be (so it touches the next window).